### PR TITLE
`_score` should not be a reserved attribute in ES|QL 

### LINKED ISSUE #118460: allow _score in ES|QL to be manipulated
make it possible to override and manipulate `_score` attributes / fields in ES|QL.

### PR DESCRIPTION
**Pull request by tteofili**
Drop checks in ES|QL `Verifier` that ensured `_score` is a reserved attribute.
These checks are too restrictive, in addition to that relaxing them would make handling of `_score` consistent with other (metadata) attributes and fields.

fixes #118460

### CODE REVIEW

**Review by ioanatia — COMMENTED** [review]
The changes look right with me!
Can we get a CSV test just to prevent any future regressions with allowing `_score` to be overwritten?
Something like what you had in the test before: `"from foo metadata _score | where qstr(\"bar\") | eval _score = _score + 1"`
and another one where `_score` receives a different type:

`from foo metadata _score | where qstr(\"bar\") | eval _score = "foobar"`

with that, I think this should be good to go!

**Comment by ioanatia** [comment]
also what would happen if `_score` is an actual indexed field? have we tested that?
we don't have to fix it here, but can be a follow up in the meta issue https://github.com/elastic/elasticsearch/issues/116599

**Comment by tteofili** [comment]
> also what would happen if `_score` is an actual indexed field? have we tested that?

added an item in #116599 

> an we get a CSV test just to prevent any future regressions with allowing _score to be overwritten?

added a couple of tests for score manipulation and override in [0ff6fb3](https://github.com/elastic/elasticsearch/pull/118435/commits/0ff6fb39a6a24236f9592c9125046fc96340f33e)

**Comment by ioanatia** [comment]
> > an we get a CSV test just to prevent any future regressions with allowing _score to be overwritten?

> added a couple of tests for score manipulation and override in [0ff6fb3](https://github.com/elastic/elasticsearch/pull/118435/commits/0ff6fb39a6a24236f9592c9125046fc96340f33e)

I did not realize this would actually create a problem for bwc - feel free to remove them and then we can add them later after the changes make it in 8.18.
The alternatives are:

- mute the tests you just added
- add a new `EsqlCapability` to signal that `_score` is no longer reserved - this is too heavy, let's not do it

**Review by carlosdelest — APPROVED** [review]

File: muted-tests.yml [hunk 1]
```diff
@@ -180,6 +180,12 @@ tests:
 - class: "org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT"
   method: "test {scoring.*}"
   issue: https://github.com/elastic/elasticsearch/issues/117641
+- class: "org.elasticsearch.xpack.esql.qa.mixed.MultilusterEsqlSpecIT"
```
Comment (ioanatia) [inline muted-tests.yml:183]: there is a typo here - also looks like a duplicate with the previous muted test (line 180)

**Comment by tteofili** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/118568)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

File: muted-tests.yml (hunk 1 above)
Comment (tteofili) [reply muted-tests.yml:183]: oh no! dummy PR to fix that https://github.com/elastic/elasticsearch/pull/118665

**Pull request merged**